lib/data/dpa/topology.js

```diff
@@ -84,16 +80,19 @@ function updateWeight(obj) {
 // generate weight distribution
 function genWeightDistr(metadata, obj, depth) {
     if (!obj.leaf && obj.constructor === Object) {
+        const max = Object.keys(obj).map(val =>
+            obj[val].maxFragsNb).filter(nb => nb !== undefined);
```

Comment:
Maybe you should invert the map and the filter function. I mean if you reduce the size of the array first, the second loop will be faster.